### DISCUSSION TITLE
Is it possible to differentiate between single-line and multi-line exports in "padding-line-between-statements" rule?

### DISCUSSION THREAD

**Discussion by abdusabri** [eslint/eslint#14351] [Q&A]
Hi 👋 

I was looking into using the [`padding-line-between-statements` rule](https://eslint.org/docs/rules/padding-line-between-statements) and was wondering if there is a way to differentiate between single-line and multi-line `export` statements.

I would like to enforce a blank line when using a multi-line export statement like 👇 

```JavaScript
// enforce a blank line
export function myFync() {
  // function code
}
```

However, in a case where there are a series of simple, one-line export statements like 👇 , I would like to disallow - or at least not enforce the rule

```JavaScript
// ignore the rule or disallow using a blank line
export const a = 'a';
export const b = 'b';
export const c = 'c';
```

In a config like 👇, is there a way to tweak it to achieve what I want? 

```JavaScript
'padding-line-between-statements': [
  'warn',
  { blankLine: 'always', prev: '*', next: 'function' },
  { blankLine: 'always', prev: '*', next: 'export' },
  { blankLine: 'never', prev: 'export', next: 'export' }
]
```

**Comment by chrisjcthomas** [comment]
I do not think `padding-line-between-statements` can express that distinction.

That rule matches by statement type selectors such as `export`, `function`, `const`, etc. It does not have a selector for “export declaration whose source range spans multiple lines” vs “export declaration that fits on one line”. So this config can distinguish `export` followed by `export`, but not the length/shape of each export statement.

Your example:

```js
{ blankLine: 'always', prev: '*', next: 'export' },
{ blankLine: 'never', prev: 'export', next: 'export' }
```

can say “do not put blank lines between consecutive export statements”, but it cannot say “unless the next export is a multi-line function/class/etc.”

The practical options are:

- choose one consistent style for all exports with the built-in rule
- rely on a formatter for the basic spacing and avoid enforcing this exact nuance
- write a small custom ESLint rule that checks `node.loc.start.line !== node.loc.end.line` for `ExportNamedDeclaration` / `ExportDefaultDeclaration`

For this exact policy, a custom rule is the clean route because the information you need is in the AST location data, but it is not exposed as a selector in `padding-line-between-statements`.
